sktm/db.py

```diff
@@ -409,7 +410,6 @@ def commit_series(self, patches, seriesid=None):
                 seriesid = 1 + res[0]
 
         for (pid, pname, purl, baseurl, projid, pdate) in patches:
-            sourceid = self.get_sourceid(baseurl, projid)
```

Comment:
I'm not at all sure this can be removed, as `get_sourceid` has a side effect. Could you please re-verify this can be removed? If you're still not sure or don't have time to check, let's leave it here.

Reply:
Agreed. We should probably rename that function (not necessarily in this pull), similarly how Django has a `.get_or_create()` method to fetch or create an object in the DB based on it's existence. For now, we can use a throwaway variable (`_`) which I think shoudln't get complaints from linters.

Reply:
Not going to dig too much through how it's used in the code right now, but a function "create if it doesn't exist and return the object" is a valid use case, IMHO. It is definitely cleaner than having `check if exists else create` blocks everywhere.

If somebody has time to analyze the possible values fed into the function to find out if we might get away with the create function itself (without needing to check the existence of the object) that would be great, but for now renaming the function to reflect what it does would make sense.

Reply:
I agree, renaming in lieu of refactoring is better than nothing.

Sure, those `check if exists else create` blocks would be bad. What I had in mind was create it once at the appropriate place and then assume it exists. Creating on demand makes for a difficult to understand behavior where you can't really be sure when something is going to be created. But let's return to this once we actually try to refactor it :)

Reply:
Ah, just to be sure you're aware, there is a bunch of similar functions there which would need a similar renaming/refactoring.

Reply:
Agreed with everything you said here :+1: Looks like the number of renamings needed would warrant a different pull request

Reply:
@spbnick The issue there is that the `sourceid` variable is never used in that method, and that causes a flake8 error.

Reply:
@major we get that (hence my idea for using a throwaway variable). The problem is that `get_sourceid` also creates the object in the database so removing the call altogether will break things.

Reply:
@major, yeah `get_sourceid` has the side effect of creating a "source" in the database, if it doesn't exist. We don't have a good way of testing sktm right now, and nobody is sure how it operates. So, we either prove removing this call doesn't break anything, by reading through the code (possibly hard), or simply leave it here for now and workaround the flake8 error (easy). Take your pick :)